tests/integration/test_update_analysis.py

```diff
@@ -0,0 +1,81 @@
+import json
+from flask.testing import FlaskClient
+from http import HTTPStatus
+
+from trailblazer.constants import TrailblazerStatus
+from trailblazer.server.schemas import AnalysisUpdateRequest
+from trailblazer.store.models import Analysis
+
+TYPE_JSON = "application/json"
+
+
+def test_update_analysis_status(client: FlaskClient, analysis: Analysis):
+    # GIVEN an analysis with a pending status
+
+    # GIVEN a valid request to set the status to completed
+    request = AnalysisUpdateRequest(status=TrailblazerStatus.COMPLETED)
+    data = request.model_dump_json()
+
+    # WHEN updating the analysis to be completed
+    response = client.put(f"/api/v1/analyses/{analysis.id}", data=data, content_type=TYPE_JSON)
+
+    # THEN it gives a success response
+    assert response.status_code == HTTPStatus.OK
+
+    # THEN it should return the analysis
+    assert response.json["id"] == analysis.id
+
+    # THEN it should have the new status
+    assert response.json["status"] == TrailblazerStatus.COMPLETED
+
+
+def test_update_analysis_comment(client: FlaskClient, analysis: Analysis):
+    # GIVEN an analysis
+
+    # GIVEN a valid request to set the comment
+    new_comment = "new comment"
```

Comment:
I think I added a fixture for this.

Reply:
I think it is more readable if defined here